doc/client_lua_api.txt

```diff
@@ -0,0 +1,812 @@
+    * `indent`: cell contents are a number and define indentation of following cell
+    * `tree`:   same as indent, but user can open and close subtrees (treeview-like)
+* Column options:
+    * `align=<value>`
+        * for `text` and `image`: content alignment within cells.
+          Available values: `left` (default), `center`, `right`, `inline`
+    * `width=<value>`
+        * for `text` and `image`: minimum width in em (default: `0`)
+        * for `indent` and `tree`: indent width in em (default: `1.5`)
+    * `padding=<value>`: padding left of the column, in em (default `0.5`).
+      Exception: defaults to 0 for indent columns
+    * `tooltip=<value>`: tooltip text (default: empty)
+    * `image` column options:
+        * `0=<value>` sets image for image index 0
+        * `1=<value>` sets image for image index 1
+        * `2=<value>` sets image for image index 2
+        * and so on; defined indices need not be contiguous empty or
+          non-numeric cells are treated as `0`.
+    * `color` column options:
+        * `span=<value>`: number of following columns to affect (default: infinite)
+
+**Note**: do _not_ use a element name starting with `key_`; those names are reserved to
+pass key press events to formspec!
+
+Spatial Vectors
+---------------
+* `vector.new(a[, b, c])`: returns a vector:
+    * A copy of `a` if `a` is a vector.
+    * `{x = a, y = b, z = c}`, if all `a, b, c` are defined
+* `vector.direction(p1, p2)`: returns a vector
+* `vector.distance(p1, p2)`: returns a number
+* `vector.length(v)`: returns a number
+* `vector.normalize(v)`: returns a vector
+* `vector.floor(v)`: returns a vector, each dimension rounded down
+* `vector.round(v)`: returns a vector, each dimension rounded to nearest int
+* `vector.apply(v, func)`: returns a vector
+* `vector.equals(v1, v2)`: returns a boolean
+
+For the following functions `x` can be either a vector or a number:
+
+* `vector.add(v, x)`: returns a vector
+* `vector.subtract(v, x)`: returns a vector
+* `vector.multiply(v, x)`: returns a scaled vector or Schur product
+* `vector.divide(v, x)`: returns a scaled vector or Schur quotient
+
+Helper functions
+----------------
+* `dump2(obj, name="_", dumped={})`
+     * Return object serialized as a string, handles reference loops
+* `dump(obj, dumped={})`
+    * Return object serialized as a string
+* `math.hypot(x, y)`
+    * Get the hypotenuse of a triangle with legs x and y.
+      Useful for distance calculation.
+* `math.sign(x, tolerance)`
+    * Get the sign of a number.
+      Optional: Also returns `0` when the absolute value is within the tolerance (default: `0`)
+* `string.split(str, separator=",", include_empty=false, max_splits=-1,
+* sep_is_pattern=false)`
+    * If `max_splits` is negative, do not limit splits.
+    * `sep_is_pattern` specifies if separator is a plain string or a pattern (regex).
+    * e.g. `string:split("a,b", ",") == {"a","b"}`
+* `string:trim()`
```

Comment:
Actually, I think those (both line 639 and 645) are correct.
Because in those cases they are actually not applying to `string` as a variable but to the string passed as first argument (`a:f()` is just a shorthand for `a.f(a)` after all).

In any case, this is very confusing. Probably it would be better to use `str` instead of `string` when using the colon (e.g. "`str:trim()`"). Or maybe try to stick to using the same syntax everywhere if possible.